### PR TITLE
Validations and Model specs

### PR DESCRIPTION
**Pull request by Victoria129**
In this PR, I have implemented the following:

- [ ]  Add the following validations:
- [ ] For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- [ ] For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- [ ] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Victoria129, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- `User` model has validations 👍 
- `Post` model has validations 👍 
- Has Unit test 👍 

## Required Changes ♻️
- [ ] Hello, Can you work on the unit test? I am asking this because it is part of the requirement

Capture
NB: This comment above is not inline to the code because you did not make the concerned file visible in this PR

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Victoria129 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] I am adding this recommendation here because it is not in the file changes! Great effort working on the last recommendation from the previous reviewer, however, while I was retesting the Model specs Validations, one test case failed.
Screenshot 2023-09-23 at 17 45 48
- Let's resolve this before asking for another review.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -1,56 +1,29 @@
...
+  let!(:comment2) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment3) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment4) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+  let!(:comment5) { Comment.create(post_id: post.id, user_id: user.id, text: 'Sample Comment') }
+
+  describe 'validations' do
+    it { should validate_presence_of(:title) }
+    it { should validate_length_of(:title).is_at_most(250) }
+    it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+    it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
```
Comment (VICTOR-CHUKWU) [inline spec/models/post_spec.rb]: - [ ] You have all test in post_spec failing due to method validate_presence, kindly fix this
image

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Victoria129 , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Victoria129 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 
- LInters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- All failing tests passing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**